### DISCUSSION TITLE
Laravel vapor support

### DISCUSSION THREAD

**Discussion by BoyeMagnus** [spatie/laravel-google-fonts#3] [Q&A] [answered]
Currently, the package does not support Laravel Vapor. It would be a nice addition :)
The main issues is that:
 - Vapor wants assets stored in the public folder (from my testing symlinks doesn't work)
 - Vapor uses the asset() function for deciding where to fetch the asset from (CloudFront / primary domain).

More about how vapor assets work can be found here:
https://docs.vapor.build/1.0/projects/deployments.html#assets

**Comment by freekmurze** [comment]
Currently, we don't need Vapor support ourselves.

Feel free to PR improvements.

**Comment by sebastiandedeyne** [comment]
Shouldn't this work out of the box if you use a different filesystem disk like S3? We're injecting a `Filesystem` object with CDN support in mind.

https://github.com/spatie/laravel-google-fonts/blob/c53f405bcb4969aa7ede247958dfe3157b1c9619/src/GoogleFonts.php#L14

**Comment by lovenick** [comment, marked as answer]
Can confirm, this does work out of the box. I published the config and changed the disk to a "public" s3 bucket. Then ran the prefetch command and it worked. The key is, you need to add filesystem config with:

```
'visibility' => 'public'
```

In my case, I added a "s3_public" config since I don't want everything in my default s3 bucket public.

**Comment by sebastiaanluca** [comment]
Also on Vapor, but not using an S3 bucket. Would it be possible to fetch/preload the fonts (configured to download to the public/assets/fonts directory) and serve it using the `ASSET_URL` environment variable? That way the assets are saved to, cached, and optimized by Cloudfront.

I got them to save to the assets directory and moved to Cloudfront when deploying, but the package tries to download them again in the public directory, which Vapor does not support (`Impossible to create the root directory "/var/task/public/assets"`).

**Reply by sebastiandedeyne** [reply]
I'd love to support this, but don't have enough Vapor experience to really understand what this takes. I'd be happy to look into this next to a PR that fixes this issue, then we can decide what the right abstraction is.
